R-package/tests/testthat/test_basic.R

```diff
@@ -3652,6 +3652,28 @@ test_that("lgb.cv() only prints eval metrics when expected to", {
   )
 })
 
+test_that("lgb.cv() works with an already constructed dataset with integer categoricals", {
```

Comment:
Since this is testing code in `lgb.Dataset`, would you consider moving it into `test_dataset.R`? I think it's similar to this one:

https://github.com/microsoft/LightGBM/blob/6e78e697f8e27d4bbf493a0ebd9134c3254a2ecd/R-package/tests/testthat/test_dataset.R#L407